Feat/12 start itinerary

Francois389 opened the pull request:
This pull request includes several changes to improve the functionality and maintainability of the Android client application. The most important changes include adding new permissions and activities, creating a `ServiceFactory` class, and updating the `Client` class to use a new `ClientCategory` enum.

### Player

Now we can see a map with a point for each client and a line that follow the road to link each client. For now, the client aren't sorted.
![image](https://github.com/user-attachments/assets/633b6e22-8cd7-4f1e-8837-d26636b028e0)

We also have a page where we can see the itinerary info and maybe create a new route.
![image](https://github.com/user-attachments/assets/7be90f94-c594-42da-ac6c-52e493ec9f5f)


### Permissions and Activities:

* [`app/src/main/AndroidManifest.xml`](diffhunk://#diff-7fa6aef292187a049f7a4d6060d8df3ba212d838789c78940bd363344b1c38cdR5-R16): Added permissions for fine and coarse location access, and added new activities for itinerary information and player. [[1]](diffhunk://#diff-7fa6aef292187a049f7a4d6060d8df3ba212d838789c78940bd363344b1c38cdR5-R16) [[2]](diffhunk://#diff-7fa6aef292187a049f7a4d6060d8df3ba212d838789c78940bd363344b1c38cdR36-R37)

### Service Factory:

* [`app/src/main/java/fr/iut_rodez/pathpilot_android_client/ServiceFactory.java`](diffhunk://#diff-5b3cdba323fea1fdb52e12dbb09ae5b13a6c9bd7adcb296b3f000094a8ad89c9R1-R30): Introduced a `ServiceFactory` class to manage service instances, starting with `IRouteService`.

### Client Class Updates:

* [`app/src/main/java/fr/iut_rodez/pathpilot_android_client/home/clients/Client.java`](diffhunk://#diff-3d086ea715fc42f66ba166d0817752ad09de2085299314dd7c1359194309244fR13-R20): Updated the `Client` class to use a new `ClientCategory` enum, added methods for address display name, and created a method to generate a client from a short JSON object. [[1]](diffhunk://#diff-3d086ea715fc42f66ba166d0817752ad09de2085299314dd7c1359194309244fR13-R20) [[2]](diffhunk://#diff-3d086ea715fc42f66ba166d0817752ad09de2085299314dd7c1359194309244fL29-R39) [[3]](diffhunk://#diff-3d086ea715fc42f66ba166d0817752ad09de2085299314dd7c1359194309244fL54-R57) [[4]](diffhunk://#diff-3d086ea715fc42f66ba166d0817752ad09de2085299314dd7c1359194309244fL66-R70) [[5]](diffhunk://#diff-3d086ea715fc42f66ba166d0817752ad09de2085299314dd7c1359194309244fL90-R94) [[6]](diffhunk://#diff-3d086ea715fc42f66ba166d0817752ad09de2085299314dd7c1359194309244fL145-R153) [[7]](diffhunk://#diff-3d086ea715fc42f66ba166d0817752ad09de2085299314dd7c1359194309244fR196-R209) [[8]](diffhunk://#diff-3d086ea715fc42f66ba166d0817752ad09de2085299314dd7c1359194309244fR223) [[9]](diffhunk://#diff-3d086ea715fc42f66ba166d0817752ad09de2085299314dd7c1359194309244fL223-R242) [[10]](diffhunk://#diff-3d086ea715fc42f66ba166d0817752ad09de2085299314dd7c1359194309244fL246-R274)

### Client Category Enum:

* [`app/src/main/java/fr/iut_rodez/pathpilot_android_client/home/clients/ClientCategory.java`](diffhunk://#diff-9b2eaf0ad5776e6ed040c9637a6a0a7ed1bbc124d521462bb607e387e383976dR1-R67): Added a new `ClientCategory` enum to represent client categories, with methods to parse from JSON and strings.

### Miscellaneous Changes:

* [`app/build.gradle`](diffhunk://#diff-51a0b488f963eb0be6c6599bf5df497313877cf5bdff3950807373912ac1cdc9R70): Added a new dependency for `osmbonuspack`.
* [`app/src/main/java/fr/iut_rodez/pathpilot_android_client/home/Home.java`](diffhunk://#diff-01ef66333cbcb71c1325283add7d81746e01a5dba6be761b8aacc50c6001fe39L16): Removed an unused import and refactored code for better readability. [[1]](diffhunk://#diff-01ef66333cbcb71c1325283add7d81746e01a5dba6be761b8aacc50c6001fe39L16) [[2]](diffhunk://#diff-01ef66333cbcb71c1325283add7d81746e01a5dba6be761b8aacc50c6001fe39R29-R37) [[3]](diffhunk://#diff-01ef66333cbcb71c1325283add7d81746e01a5dba6be761b8aacc50c6001fe39L66-R74) [[4]](diffhunk://#diff-01ef66333cbcb71c1325283add7d81746e01a5dba6be761b8aacc50c6001fe39R130)
* [`app/src/main/java/fr/iut_rodez/pathpilot_android_client/home/clients/AddClient.java`](diffhunk://#diff-a058bd8deabbb0a07228af42f2c15edb917c56b05e828ec17ac45bd271e7a0e9L69): Removed a call to `EdgeToEdge.enable(this)` in the `onCreate` method.
* [`app/src/main/java/fr/iut_rodez/pathpilot_android_client/home/clients/ClientArrayAdapter.java`](diffhunk://#diff-a29e319bd77a0b592b318b9006589c13975f3bb4256d1c5f08f0330be1c24916R49-R53): Updated to use the new address display name method.
* [`app/src/main/java/fr/iut_rodez/pathpilot_android_client/home/clients/ClientService.java`](diffhunk://#diff-609dd0bffbaa62ea929b72734274bb4d3dcadea6dac813e6777f2fbd1aec4b5aL55-R55): Modified to use `NetworkUtils.createAuthenticatedRequest` for client requests.

Francois389 committed:
feat: create view for Itinerary Info

Francois389 committed:
feat: add InfoItinerary activity

- Created a new activity to display itinerary details.
- Integrated client list with an adapter for better UI.
- Added buttons for starting the itinerary and going back.
- Implemented error handling for missing itinerary data.

Francois389 committed:
feat: add activity to AndroidManifest

+ fix import

Francois389 committed:
fix: the list's width match the parent

Francois389 committed:
fix: clean `FragmentClients`

Francois389 committed:
fix: also parse the compagnyLocation

Francois389 committed:
feat: make Itinerary Parcelable

Added Parcelable implementation to the Itinerary class.
- Implemented methods for parceling and unparceling data.
- Created a static CREATOR field for easy instance creation from a Parcel.
- Updated writeToParcel and describeContents methods for proper serialization.

Francois389 committed:
feat: add redirection to `InfoItinerary`

Francois389 committed:
fix: clean import

Francois389 committed:
feat: add back icon to AddItinerary activity

Francois389 committed:
feat: add CurrentPosition class for location handling

- Implemented a new class to manage GPS location.
- Added methods for requesting and checking location permissions.
- Included functionality to retrieve the current position and convert it to GeoPoint.
- Handled permission request results with callbacks.

Francois389 committed:
feat: add current position handling to map selection

- Introduced CurrentPosition for managing location permissions.
- Updated map center logic to use current position if no point is given.
- Enhanced permission request handling in onRequestPermissionsResult.

Francois389 committed:
fix: extracted method to improve clarity

Francois389 committed:
feat: add display name for itinerary

Francois389 committed:
feat: add icon

Francois389 committed:
fix: the map height is correctly set

Francois389 committed:
feat: add player itinerary layout

Created a new XML layout for the player itinerary screen. Key features include:
- Header with client details and navigation buttons
- Counter for visited clients
- Integrated map view
- Action buttons for various functionalities like stop, pause, and settings

Francois389 committed:
feat: add Callable if access is denied

Francois389 committed:
feat: create `PlayerItinerary` activity

Francois389 committed:
fix: remove edgeToEdge

Francois389 committed:
feat: add GeoPoint method to Client class

- Introduced a new method to get GeoPoint.
- Returns a GeoPoint object using latitude and longitude.

Francois389 committed:
feat: add PlayerItinerary activity

Added a new activity for managing player itineraries in the app. This enhances navigation and functionality related to itinerary management.

Francois389 committed:
feat: add string

Francois389 committed:
fix: the list of added client height

Francois389 committed:
feat: add addressName as a property

Francois389 committed:
feat: simplify client address handling

- Updated client address retrieval to use display name.
- Removed unnecessary imports and cleaned up code.
- Streamlined dropdown view for client selection.

Francois389 committed:
feat: add info of the nextClient

Francois389 committed:
feat: update client handling in itinerary info

- Added logic to set address display name for clients.
- Updated client list retrieval before adapter initialization.
- Improved logging for client setup process.

Francois389 committed:
feat: add warning string

Francois389 committed:
feat: update French strings for location permissions

Updated the French translations for location permission messages to improve clarity and user experience.

Francois389 committed:
feat: update CurrentPosition to use the location from OSM

Francois389 committed:
fix: clean file

Francois389 committed:
feat: update button text

Francois389 committed:
feat: rename method to to better match the behaviour

Francois389 committed:
feat: create RouteService

Francois389 committed:
feat: add Route class for managing sales routes

- Introduced a new Route class implementing Parcelable.
- Added properties for salesman home, expected and visited clients, start date, current client index, paused state, and current position.
- Implemented parceling methods to save and restore route data.

Francois389 committed:
feat: implement `IRouteService`

Francois389 committed:
feat: add ServiceFactory

Francois389 committed:
feat: add base API URL for route service

- Introduced a constant for the base API URL.
- Added a TODO comment in the createRoute method.

Francois389 committed:
fix: address name multiline

Francois389 committed:
feat: add displayed information

Francois389 committed:
feat: update MapSelection

Francois389 committed:
feat: simplify network requests with auth utility

Refactored network request creation to use a utility method for authentication. Removed redundant header setup in two service classes, streamlining the code and improving readability.

Francois389 committed:
feat: add client creation from short JSON object

- Implemented a method to create a Client from a simplified JSON structure.
- Updated Itinerary class to utilize the new client creation method.
- Refactored salesman location handling to use GeoPoint directly.
- Cleaned up some parsing methods in the Parser utility.

Francois389 committed:
feat: WIP add route management methods

- Implemented clientVisited method to notify server of visited clients.
- Added getCurrentClient and clientHasBeenVisited methods for route updates.
- Enhanced createRoute with documentation for clarity.

Francois389 committed:
feat: refactor client category handling

- Changed clientCategory from String to ClientCategory enum.
- Updated constructor and methods to use the new enum type.
- Added parsing methods for JSON and string conversions in the new ClientCategory class.
- Simplified category retrieval and setting logic throughout the code.

Francois389 committed:
test: add test to parse a Route

Francois389 committed:
feat: implement test of Parser

- Changed expected and visited clients from arrays to ArrayLists.
- Added a new constructor in Route to parse JSON data.
- Updated methods in Route to work with ArrayLists instead of arrays.
- Introduced utility methods in Parser for parsing routes and short clients from JSON.

Francois389 committed:
feat: add JWT token handling in itinerary flow

- Pass JWT token from itinerary list to details screen
- Implement method to retrieve and set up the token
- Show error dialog if token is missing

Francois389 committed:
feat: add progress dialog and error handling

- Introduced a progress dialog for better user feedback.
- Added methods to show and dismiss the progress dialog.
- Implemented an error dialog method for displaying error messages.

Francois389 committed:
feat: update route service for itinerary creation

- Changed API endpoint to use base URL.
- Updated createRoute method to accept activity and itinerary parameters.
- Revised Javadoc comments for clarity on functionality.

Francois389 committed:
chore: extract `Button` to a new file `DialogButton`

that way I can add default button so the creation is faster

Francois389 committed:
feat: Add route creation and navigation logic

- Implemented route creation from an itinerary.
- Added method to redirect to player activity with the new route.
- Updated token handling and error messaging for better user experience.

Francois389 committed:
chore: add osmbonuspack dependency

- Included osmbonuspack library for enhanced map features.
- Updated version reference in the libs.versions.toml file.
- Added JitPack repository to settings.gradle for dependency resolution.

Francois389 committed:
fix: latitude and longitude was swaped

Francois389 committed:
feat: add LocationNameProvider for address retrieval

Introduced a new class to fetch address names from GeoPoints. It uses Geocoder to get the address and handles cases where the address isn't found, returning a default string resource instead.

Francois389 committed:
feat: refactor dialog button listeners

- Extracted dismiss and finish logic into separate methods.
- Simplified button creation for better readability.
- Improved code organization by using dedicated listener methods.

Francois389 committed:
feat: enhance route management and map display

- Added road manager for routing functionality.
- Implemented methods to set route information and draw polylines on the map.
- Improved client marker handling with dedicated methods.
- Updated UI elements to reflect next client details dynamically.
- Refactored variable names for clarity regarding route status.

Francois389 committed:
feat: enhance route display and client markers

- Added loading popup during route drawing.
- Improved polyline outline with custom stroke width and color.
- Updated client marker method to include index in labels.
- Dismissed loading dialog on completion or error.

Francois389 committed:
fix: reorder the method so the marker are above the line

Francois389 committed:
fix: remove unused parameter

Francois389 committed:
fix: fiw sonarcloud issue

Francois389 committed:
chore: move Popup to popup package

Francois389 committed:
fix: rename function name to be camelCase

Francois389 committed:
refactor: change return type to List for flexibility

Updated the return type of getShortClients method from ArrayList to List. This enhances flexibility and allows for better abstraction in handling client lists.

Francois389 committed:
refactor: move LocationNameProvider to util package

- Changed the package of LocationNameProvider from map to util.
- Updated import statements in related files.
- Made constructor private for singleton pattern.

Francois389 committed:
fix: fill ignored exception catch

Francois389 committed:
feat: use constant for company name JSON key

- Replaced hardcoded "companyName" string with a constant.
- Added a private static class to hold the JSON key constant.
- Updated methods to reference the new constant for better maintainability.

Francois389 committed:
refactor: make ClientConstant public

Changed ClientConstant from private to public for better access.

Francois389 committed:
fix: use currentPosition only if permission is granted

Moved the call to setNextClientInfo back into the flow after permission is granted. This ensures that the next client information is updated correctly when the user allows location access.

Francois389 committed:
refactor: change return types to ArrayList

Updated methods to return ArrayList instead of List.
- getClientsPageable now returns ArrayList<Client>
- getShortClients updated to return ArrayList<Client>
- getItinerariesPageable changed to return ArrayList<Itinerary>

quentinformatique commented on app/src/main/java/fr/iut_rodez/pathpilot_android_client/ServiceFactory.java:
missing javadoc

Francois389 replied:
Fix in 776945924f67416187a64d36b754a3990646f873

quentinformatique commented on app/src/main/java/fr/iut_rodez/pathpilot_android_client/home/itinerary/InfoItinerary.java:
```diff
@@ -0,0 +1,116 @@
+
+        if (itinerary == null) {
+            Log.e(TAG, "onCreate: No itinerary found in the intent");
+            popup.showAlertDialog("Error", "No itinerary found in the intent"); // TODO i18n
```
complete this TODO

Francois389 replied:
Fix in 7ae49bec2e4a93ec3dcb10f200e29c475cbb6e02

quentinformatique commented on app/src/main/java/fr/iut_rodez/pathpilot_android_client/home/itinerary/InfoItinerary.java:
```diff
@@ -0,0 +1,116 @@
+        } else {
+            // This should never happen
+            Log.e(TAG, "setUpToken: No token found in the intent");
+            popup.showAlertDialog("Error", "No token found in the intent", new DialogButton("Ok", (dialog, which) -> finish()), null, null); // TODO i18n
```
complete this TODO

Francois389 replied:
Fix in 7ae49bec2e4a93ec3dcb10f200e29c475cbb6e02

quentinformatique commented on app/src/main/java/fr/iut_rodez/pathpilot_android_client/home/itinerary/PlayerItinerary.java:
```diff
@@ -0,0 +1,308 @@
+
+        if (route == null) {
+            Log.e(TAG, "onCreate: No itinerary found in the intent");
+            popup.showAlertDialogOK("Error", "No itinerary found in the intent", DialogButton.okFinish(this)); // TODO i18n
```
complete this TODO

Francois389 replied:
Fix in 7ae49bec2e4a93ec3dcb10f200e29c475cbb6e02

quentinformatique commented on app/src/main/java/fr/iut_rodez/pathpilot_android_client/home/itinerary/PlayerItinerary.java:
```diff
@@ -0,0 +1,308 @@
+            // if the road build process failed, show an error dialog
+            if (road.mStatus != Road.STATUS_OK) {
+                Log.e(TAG, "setRoutePolyline: Error while drawing the road");
+                popup.showAlertDialogOK("Error", "Error while drawing the road", DialogButton.okDismiss()); // TODO i18n
```
complete this TODO

Francois389 replied:
Fix in 7ae49bec2e4a93ec3dcb10f200e29c475cbb6e02

quentinformatique commented on app/src/main/java/fr/iut_rodez/pathpilot_android_client/home/routes/RouteService.java:
```diff
@@ -0,0 +1,69 @@
+                },
+                error -> {
+                    popup.dismissProgressDialog();
+                    // TODO better message
```
Complete this TODO

Francois389 replied:
Fix in 358b84e18f2ac312946a0d448f1af73cddaa5bd0

quentinformatique commented on app/src/main/java/fr/iut_rodez/pathpilot_android_client/signup/SignUpService.java:
```diff
@@ -59,7 +60,7 @@ public static void signUp(SignUpInput signUpInput, Context context) {
 
                     // Button to go to login page
-                    Popup.Button btnLogin = new Popup.Button("Login", (dialog, which) -> {
+                    DialogButton btnLogin = new DialogButton("Login", (dialog, which) -> {
```
hard coded string

Francois389 replied:
Fix in 00b6b77aff36ab829926aa6b3153212bafa9e6da

quentinformatique commented on app/src/main/java/fr/iut_rodez/pathpilot_android_client/signup/SignUpService.java:
```diff
@@ -68,7 +69,7 @@ public static void signUp(SignUpInput signUpInput, Context context) {
                     });
                     // Button to dismiss the dialog
-                    Popup.Button btnOk = new Popup.Button("OK", (dialog, which) -> dialog.dismiss());
+                    DialogButton btnOk = new DialogButton("OK", (dialog, which) -> dialog.dismiss());
```
hard coded string

Francois389 replied:
Fix in 00b6b77aff36ab829926aa6b3153212bafa9e6da

quentinformatique commented on app/src/main/java/fr/iut_rodez/pathpilot_android_client/signup/SignUpService.java:
```diff
@@ -68,7 +69,7 @@ public static void signUp(SignUpInput signUpInput, Context context) {
+                    DialogButton btnOk = new DialogButton("OK", (dialog, which) -> dialog.dismiss());
 
                     // Show success dialog with the two buttons
                     popup.showAlertDialog("Success", "Account created successfully", btnLogin, btnOk, null);
```
hard coded string

Francois389 replied:
Fix in 00b6b77aff36ab829926aa6b3153212bafa9e6da

quentinformatique commented on app/src/main/java/fr/iut_rodez/pathpilot_android_client/util/popup/DialogButton.java:
```diff
@@ -0,0 +1,53 @@
+     * @return the button
+     */
+    public static DialogButton okDismiss() {
+        return new DialogButton("OK", getDismissListener());
```
hard coded string

Francois389 replied:
For this one the string will be the same in French and in English.
So I didn't create a string. 

Should I ?

quentinformatique replied:
menfou

quentinformatique replied:
It could be a better practice to do this it even if it's useless, but you can live it as it is because it doesn't impact the user.

Francois389 replied:
Fix in 00b6b77aff36ab829926aa6b3153212bafa9e6da

quentinformatique commented on app/src/main/java/fr/iut_rodez/pathpilot_android_client/util/popup/DialogButton.java:
```diff
@@ -0,0 +1,53 @@
+     * @return the button
+     */
+    public static DialogButton okFinish(Activity activity) {
+        return new DialogButton("OK", getFinishListener(activity));
```
hard coded string

Francois389 replied:
Fix in 00b6b77aff36ab829926aa6b3153212bafa9e6da

Francois389 committed:
feat: add i18n support for error messages

- Updated error handling to use localized strings.
- Replaced hardcoded error messages with string resources.
- Added new string resources for various error scenarios.

Francois389 committed:
feat: improve error handling in route creation

- Added VolleyErrorHandler for better error messages.
- Replaced hardcoded error dialog with a more flexible handler.

quentinformatique commented on app/src/main/java/fr/iut_rodez/pathpilot_android_client/util/popup/Popup.java:
```diff
@@ -0,0 +1,111 @@
+     * @param message the message of the dialog
+     */
+    public void showAlertDialog(String title, String message) {
+        DialogButton okDialogButton = new DialogButton("OK", (dialog, which) -> dialog.dismiss());
```
hard coded string

Francois389 replied:
Fix in 00b6b77aff36ab829926aa6b3153212bafa9e6da

quentinformatique commented on app/src/main/res/layout/view_create_itinerary.xml:
```diff
@@ -6,12 +6,12 @@
     xmlns:app="http://schemas.android.com/apk/res-auto">
     <!-- TODO write the right title "New itinerary" -->
```
complete this todo

Francois389 replied:
Fix in 8af643317f0f71c1fd1dbe1fb91291b48f52200b

quentinformatique commented on app/src/main/res/layout/view_create_itinerary.xml:
```diff
@@ -6,12 +6,12 @@
-    <include layout="@layout/component_header_no_text" />
+    <include layout="@layout/component_header_back_icon" />
 
     <!-- TODO faire le vrai footer -->
```
complete this todo

Francois389 replied:
Fix in 8af643317f0f71c1fd1dbe1fb91291b48f52200b

quentinformatique commented on app/src/main/res/layout/view_player_itinerary.xml:
```diff
@@ -0,0 +1,178 @@
+                android:layout_width="wrap_content"
+                android:layout_height="wrap_content"
+                android:textColor="@color/white"
+                tools:text="Client Name" />
```
use strings file

Francois389 replied:
This string won't be in the built view. With `tools:text` the given text will only be visible in Android Studio.

quentinformatique commented on app/src/main/res/layout/view_player_itinerary.xml:
```diff
@@ -0,0 +1,178 @@
+                android:layout_width="wrap_content"
+                android:layout_height="wrap_content"
+                android:textColor="@color/white"
+                tools:text="10km" />
```
use strings file

quentinformatique commented on app/src/main/res/layout/view_player_itinerary.xml:
```diff
@@ -0,0 +1,178 @@
+            app:layout_constraintStart_toEndOf="@id/separator"
+            app:layout_constraintEnd_toStartOf="@id/back_btn"
+            app:layout_constraintTop_toTopOf="parent"
+            tools:text="Client Address (Rodez Intermarche Gineste, 94 Av. de la Gineste, 12000 Rodez, France)" />
```
use strings file

quentinformatique commented on app/src/main/res/layout/view_player_itinerary.xml:
```diff
@@ -0,0 +1,178 @@
+        app:layout_constraintTop_toBottomOf="@+id/header"
+        app:layout_constraintEnd_toEndOf="parent"
+        app:layout_constraintStart_toStartOf="parent"
+        tools:text="Client visited 0/2" />
```
use strings file

Francois389 committed:
feat: add JavaDoc

Francois389 committed:
feat: update itinerary creation UI and logic

- Set header text for creating an itinerary.
- Added back button functionality to return to the previous screen.
- Initialized a default client in the list for better UX.
- Updated string resources for consistency across languages.

Francois389 committed:
feat: update dialog buttons and success messages

- Changed dialog button text to use string resources.
- Updated error handling to improve user feedback.
- Added new string resources for better localization.

Odonata971 commented on app/src/main/java/fr/iut_rodez/pathpilot_android_client/home/clients/ClientCategory.java:
```diff
@@ -0,0 +1,68 @@
+     * </p>
+     *
+     * @param category
+     * @return
```
missing the @return comment completed

Francois389 replied:
Fix in 738e136e43c69574796d0bdf76a578530355ce6d

Odonata971 commented on app/src/main/java/fr/iut_rodez/pathpilot_android_client/home/clients/FragmentClients.java:
```diff
@@ -115,9 +107,9 @@ public interface AddClient {
     }
 
-    public ArrayList<Client> getListClients(){
+    public ArrayList<Client> getListClients() {
```
missing comment

Francois389 replied:
Fix in 194766991aa2200bf584356f27e20c231b454c3d

Odonata971 commented on app/src/main/java/fr/iut_rodez/pathpilot_android_client/home/itinerary/Itinerary.java:
```diff
@@ -13,57 +15,68 @@
-                    clientCoordinates.getDouble("y")
-            );
-            this.clients.add(client);
+            clientsParsed.add(Client.createClientFromShortJson(clientJson));
```
I like this

Odonata971 commented on app/src/main/java/fr/iut_rodez/pathpilot_android_client/home/itinerary/PlayerItinerary.java:
```diff
@@ -0,0 +1,308 @@
+    }
+
+
+    private void initialiseMap() {
```
missing javadoc

Francois389 replied:
Fix in b6c5ff0675bb45003544463baeb4c5375684b84c

Odonata971 commented on app/src/main/res/drawable/icon_pause.xml:
```diff
@@ -0,0 +1,12 @@
+      android:pathData="M10,0L11,0A10,10 0,0 1,21 10L21,104.5A10,10 0,0 1,11 114.5L10,114.5A10,10 0,0 1,0 104.5L0,10A10,10 0,0 1,10 0z"
+      android:fillColor="#ffffff"/>
+  <path
+      android:pathData="M71,0L72,0A10,10 0,0 1,82 10L82,104.5A10,10 0,0 1,72 114.5L71,114.5A10,10 0,0 1,61 104.5L61,10A10,10 0,0 1,71 0z"
```
error line 10:73

Odonata971 commented on app/src/main/res/values-fr/strings.xml:
```diff
@@ -14,6 +14,8 @@ String for Login activity
     <!--
     String for Sign up activité
     -->
+    <string name="account_created_successfully">Creation du compte reussie</string>
```
Where are the accents ?

Francois389 replied:
There are retired

Francois389 replied:
Fix in babb0f0d741ebe080ea9becde54b80f84dc0949d

Odonata971 commented on app/src/main/res/values-fr/strings.xml:
```diff
@@ -99,10 +104,12 @@ String for Login activity
     <string name="error_min_clients_per_itinerary">Vous devez ajouter au moins 1 client dans un itinéraire pour le créer</string>
     <string name="title_list_clients_added_for_itinerary"><u>Clients ajoutés</u></string>
     <string name="button_create_itinerary">Créer</string>
+    <string name="header_create_itinerary">Create itinerary</string>
```
English in french strings?

Francois389 replied:
Fix in babb0f0d741ebe080ea9becde54b80f84dc0949d

Odonata971 commented on app/src/main/res/values-fr/strings.xml:
```diff
@@ -113,11 +120,16 @@ String for Login activity
     Map selection
     -->
-    <string name="validate_selected_position">Valider la position sélectionnée</string>
+    <string name="address_unknown">Adresse inconnu</string>
```
Missing the "e" at the end of inconnu

Francois389 replied:
Fix in babb0f0d741ebe080ea9becde54b80f84dc0949d

Odonata971 commented on app/src/main/res/values-fr/strings.xml:
```diff
@@ -134,4 +146,17 @@ String for Login activity
+    <!--
+    Itinerary Player
+    -->
+    <string name="need_to_allow_location_permission">Vous devez autoriser l\'accés à la localisation pour utiliser la carte.</string>
```
Grave accent "è" instead of "é"

Francois389 replied:
Fix in babb0f0d741ebe080ea9becde54b80f84dc0949d

Francois389 committed:
docs: update JSON parsing documentation

Added example JSON format for clarity and improved parameter description in the fromJSON method.

Francois389 committed:
feat: add JavaDoc to getListClients

Francois389 committed:
feat: add JavaDoc to initialiseMap

Francois389 committed:
fix: french typos

Odonata971 commented on app/src/main/res/values-fr/strings.xml:
```diff
@@ -73,6 +75,7 @@ String for Login activity
     <string name="confirm_password_not_same">Écrivez une seconde fois votre mot de passe pour confirmer la saisie</string>
     <string name="address_missing">Selectionnez votre position</string>
```
Missing "é" in "Selectionnez"

Francois389 committed:
fix: french typos

Odonata971 reviewed (approved):
LGTM

quentinformatique reviewed (approved):
Ok
